lib/services/localization_service.dart

```diff
@@ -173,11 +189,36 @@ class LocalizationService {
   /// Get device locale or fallback
   static String getDeviceLocale() {
     try {
-      final deviceLocale = WidgetsBinding.instance.platformDispatcher.locale;
+      final deviceLocale =
+          deviceLocaleOverride != null
+              ? deviceLocaleOverride!()
+              : WidgetsBinding.instance.platformDispatcher.locale;
+
       final localeCode = deviceLocale.languageCode;
       return _supportedLocales.containsKey(localeCode) ? localeCode : _fallbackLocale;
     } catch (e) {
       return _fallbackLocale;
     }
   }
 }
+
+extension LanguageTextExtension on String {
+  String toLanguageDisplayText() {
+    final deviceLocale = LocalizationService.getDeviceLocale();
+
+    if (this == 'system') {
+      final systemLang =
+          LocalizationService.supportedLocales[deviceLocale] ?? deviceLocale.toUpperCase();
+
+      return 'System ($systemLang)';
```

Comment:
THis System word is not translated....  we should use the `shared.system` keyword from the locales.

Reply:
There already is a shared.system key, it is not a broader change